plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/columnsSelectionDsl.kt

```diff
@@ -19,12 +31,49 @@ internal class And10 : AbstractInterpreter<ColumnsResolver>() {
     }
 }
 
-class SingleColumnApproximation(val col: ColumnWithPathApproximation) : ColumnsResolver {
+class SingleColumnApproximation(val col: ColumnWithPathApproximation) : ColumnsResolver, SingleColumn<Any?>, ColumnReference<Any?> {
+
+    override fun name(): String {
+        return col.column.name
+    }
+
+    override fun rename(newName: String): ColumnReference<Any?> {
+        return SingleColumnApproximation(ColumnWithPathApproximation(col.path, col.column.rename(newName)))
+    }
+
     override fun resolve(df: PluginDataFrameSchema): List<ColumnWithPathApproximation> {
         return listOf(col)
     }
+
+    override fun resolve(context: ColumnResolutionContext): List<ColumnWithPath<Any?>> {
+        return listOf(resolveSingle(context))
+    }
+
+    override fun resolveSingle(context: ColumnResolutionContext): ColumnWithPath<Any?> {
+        return ColumnWithPath(col.column.asDataColumn(), col.path)
+    }
 }
 
-interface ColumnsResolver {
+interface ColumnsResolver : org.jetbrains.kotlinx.dataframe.columns.ColumnSet<Any?> {
```

Comment:
Why does this not inherit `org.jetbrains.kotlinx.dataframe.columns.ColumnsResolver<Any?>` like the name suggests? I suspect this is gonna be confusing in the future (oh it's because I sealed it... oops... oh well, just have to be careful with instance checks for when some columnsResolver is a singlecolumn or columnset)

Reply:
sounds great :)